sql_assesment.txt

```diff
@@ -0,0 +1,1786 @@
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3002 |  1900.1 | 1990-03-10 | 2007 | 1004 |
+| 3005 | 5160.45 | 1990-03-10 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-04-10 | 2002 | 1003 |
+| 3008 |    4723 | 1990-05-10 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-06-10 | 2006 | 1001 |
++------+---------+------------+------+------+
+
+
+
+
+17) Write a query that uses the EXISTS operator to extract all salespeople who have customers with a rating of 300.
+
+select * from salespeople where exists(select * from customers where snum= customers.snum and rating=300);
+
++------+---------+------------+------+
+| SNUM | SNAME   | CITY       | COMM |
++------+---------+------------+------+
+| 1001 | Peel    | London     |   12 |
+| 1002 | Serres  | SanJose    |   13 |
+| 1003 | AxelRod | New York   |   10 |
+| 1004 | Motika  | London     |   11 |
+| 1007 | Rofkin  | Barcenlona |   15 |
+| 1008 | Fran    | London     |   25 |
++------+---------+------------+------+
+
+
+
+
+
+18) Find all customers whose cnum is 1000 above the snum of Serres. 
+
+select cname,cnum from customers where cnum> 1000+(select snum from salespeople where sname= "Serres");
+
++----------+------+
+| cname    | cnum |
++----------+------+
+| Liu      | 2003 |
+| Grass    | 2004 |
+| Ciemens  | 2006 |
+| Cisneros | 2008 |
+| Periera  | 2007 |
++----------+------+
+
+
+
+
+19) Give the salespeople’s commissions as percentages instead of decimal numbers. 
+
+select sname, concat(comm,"%") as Commision from salespeople;
+
++---------+-----------+
+| sname   | Commision |
++---------+-----------+
+| Peel    | 12%       |
+| Serres  | 13%       |
+| AxelRod | 10%       |
+| Motika  | 11%       |
+| Rofkin  | 15%       |
+| Fran    | 25%       |
++---------+-----------+
+
+
+
+
+20) Find the largest order taken by each salesperson on each date, eliminating those Maximum orders, which are less than 3000.
+
+select s.sname, max(o.amt), o.odate from salespeople s inner join orders o on o.snum= s.snum group by o.odate,s.sname having(max(o.amt)>3000);
+
++--------+------------+------------+
+| sname  | max(o.amt) | odate      |
++--------+------------+------------+
+| Serres |    5160.45 | 1990-03-10 |
+| Peel   |       4723 | 1990-05-10 |
+| Peel   |    9891.88 | 1990-06-10 |
++--------+------------+------------+
+
+
+
+
+
+21) List all the largest orders for October 3, for each salesperson. 
+
+select s.sname,max(o.amt) from salespeople as s, orders as o where s.snum= o.snum and o.odate= "1990-03-10" group by s.sname;
+
++--------+------------+
+| sname  | max(o.amt) |
++--------+------------+
+| Rofkin |    1098.16 |
+| Peel   |     767.19 |
+| Motika |     1900.1 |
+| Serres |    5160.45 |
++--------+------------+
+
+
+
+
+22) Find all customers located in cities where Serres has customers. 
+
+
+select city from customers where snum=(select snum from salespeople where sname="Serres");
```

Comment:
Incorrect. We have to display customers located in the cities where Serres has his customers. Please check it.